### PR TITLE
AddressSanitizer fixes

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Some more fixes for bugs exposed by AddressSanitizer and LeakSanitizer. Mostly trivial stuff in this batch.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch3/tcp

**Comment by raffenet** [comment]
test:jenkins/ch3/tcp

**Comment by raffenet** [comment]
Added a commit to fix another leak at the hydra proxy level.
test:jenkins/ch3/tcp

**Comment by raffenet** [comment]
test:jenkins/ch4/most

File: src/pm/hydra/pm/pmiserv/pmip_cb.c [hunk 1]
```diff
@@ -93,6 +93,77 @@ static HYD_status stdoe_cb(int fd, HYD_event_t events, void *userp)
...
+    HYDU_ERR_POP(status, "unable to send PMI command upstream\n");
+    HYDU_ASSERT(!closed, status);
+
+  fn_exit:
+    MPL_free(pmi_cmd);
+    if (args) {
+        HYDU_free_strlist(args);
+        MPL_free(args);
+    }
+    MPL_free(buf);
```
Comment (hzhou) [inline src/pm/hydra/pm/pmiserv/pmip_cb.c]: I am not sure freeing a buffer that is being passed to is a good pattern. Ideally, it should be freed in the same function that allocates it; or with a clear comment indicate the transfer of ownership. Can we free it in `pmi_cb`?

Comment (raffenet) [reply src/pm/hydra/pm/pmiserv/pmip_cb.c]: Sure, now that you say it I agree it is a bad pattern. Will push new version shortly.

**Comment by raffenet** [comment]
test:jenkins/ch4/most

**Pull request merged**